src/network/protocol/messages/filteradd.zig

```diff
@@ -0,0 +1,78 @@
+const std = @import("std");
+const protocol = @import("../lib.zig");
+
+/// FilterAddMessage represents the "filteradd" message
+///
+/// https://developer.bitcoin.org/reference/p2p_networking.html#filteradd
+pub const FilterAddMessage = struct {
+    element: []const u8,
+
+    const Self = @This();
+
+    pub fn name() *const [12]u8 {
+        return protocol.CommandNames.FILTERADD ++ [_]u8{0} ** 3;
+    }
+
+    /// Returns the message checksum
+    pub fn checksum(self: *const Self) [4]u8 {
+        var digest: [32]u8 = undefined;
+        std.crypto.hash.sha2.Sha256.hash(self.element, &digest, .{});
+        std.crypto.hash.sha2.Sha256.hash(&digest, &digest, .{});
+        return digest[0..4].*;
+    }
+
+    /// Serialize the message as bytes and write them to the Writer.
+    pub fn serializeToWriter(self: *const Self, w: anytype) !void {
+        try w.writeAll(self.element);
```

Comment:
comptime {
        if (!std.meta.hasFn(@TypeOf(w), "writeAll")) @compileError("Expects r to have fn 'writeAll'.");

Reply:
added this above `try w.writeAll(self.element);`